file-storage.js

```diff
@@ -0,0 +1,28 @@
+// The Following code t
+const contactForm = document.querySelector('form')
+const fnameVal = document.getElementById('f_name')
+const lnameVal = document.getElementById('l_name')
+const emailVal = document.getElementById('mail')
+const msgVal = document.getElementById('msg')
+
+contactForm.addEventListener('input', () => {
+  const formData = {
+    firstName: fnameVal.value,
+    lastName: lnameVal.value,
+    email: emailVal.value,
+    message: msgVal.value
+  }
+  localStorage.setItem('userData', JSON.stringify(formData))
+})
+
+const object = localStorage.getItem('userData')
+const getValue = JSON.parse(object)
+
+window.addEventListener('load', () => {
+  if (localStorage.getItem('userData')) {
+    document.getElementById('f_name').value = getValue.firstName
+    document.getElementById('l_name').value = getValue.lastName
+    document.getElementById('mail').value = getValue.userEmail
```

Comment:
Great job but let's fix the small email bug. It needs to be `email`, not 'userEmail` 😎

<img width="519" alt="Screenshot 2023-02-02 at 11 06 23 PM" src="https://user-images.githubusercontent.com/25887253/216438150-537cb786-fe14-41b4-8027-421da76297af.png">